### PR TITLE
Fix broken README.md link to cli.md

### LINKED ISSUE #1021: README.md link to cli.md is broken
If you go to the the header h2 in the `README.md`:

```markdown
## Using DataFusion as a binary

DataFusion also includes a simple command-line interactive SQL utility. See the [CLI reference](datafusion/docs/cli.md) for more information.
```

The link that points to is: https://github.com/apache/arrow-datafusion/blob/master/datafusion/docs/cli.md which is broken (deleted file)

### PR DESCRIPTION
**Pull request by storopoli**
# Which issue does this PR close?

Closes #1021

# Rationale for this change
Just a quick fix for a broken link

# What changes are included in this PR?
README.md only.

# Are there any user-facing changes?
No.

### CODE REVIEW

**Comment by houqp** [comment]
Thanks @storopoli , now that our documentation is up, perhaps https://arrow.apache.org/datafusion/cli/index.html would have been a better link :)

File: README.md [hunk 1]
```diff
@@ -132,7 +132,7 @@ datafusion = "5.0.0"
 
 ## Using DataFusion as a binary
 
-DataFusion also includes a simple command-line interactive SQL utility. See the [CLI reference](datafusion/docs/cli.md) for more information.
+DataFusion also includes a simple command-line interactive SQL utility. See the [CLI reference](docs/source/cli/index.rst) for more information.
```
Comment (storopoli) [inline README.md:135]:
```suggestion
DataFusion also includes a simple command-line interactive SQL utility. See the [CLI reference](https://arrow.apache.org/datafusion/cli/index.html) for more information.
```

**Comment by storopoli** [comment]
See if it is OK now

**Comment by houqp** [comment]
Thanks @storopoli

**Comment by storopoli** [comment]
My pleasure

**Pull request merged**